xcp_d/utils/plot.py

```diff
@@ -1000,3 +1001,179 @@ def _get_tr(img):
+    entities_to_use = {f: file_entities[f] for f in file_entities if f in entities_to_use}
+    template_file = get_template(template=space, **entities_to_use, suffix='T1w', desc=None)
+    if isinstance(template_file, list):
+        template_file = template_file[0]
+    template = str(template_file)
+    output_path = os.path.abspath(output_path)
+    plott.plot_stat_map(filename,
+                        bg_img=template,
+                        display_mode='mosaic',
+                        cut_coords=8,
+                        output_file=output_path)
+    return output_path
+
+
+def surf_data_from_cifti(data, axis, surf_name):
+    """From https://neurostars.org/t/separate-cifti-by-structure-in-python/17301/2.
+
+    https://nbviewer.org/github/neurohackademy/nh2020-curriculum/blob/master/we-nibabel-markiewicz/NiBabel.ipynb
+    """
+    assert isinstance(axis, nb.cifti2.BrainModelAxis)
+    for name, data_indices, model in axis.iter_structures():
+        # Iterates over volumetric and surface structures
+        if name == surf_name:  # Just looking for a surface
+            data = data.T[data_indices]
+            # Assume brainmodels axis is last, move it to front
+            vtx_indices = model.vertex
+            # Generally 1-N, except medial wall vertices
+            surf_data = np.zeros((vtx_indices.max() + 1,) + data.shape[1:], dtype=data.dtype)
+            surf_data[vtx_indices] = data
+            return surf_data
+
+    raise ValueError(f"No structure named {surf_name}")
+
+
+def plot_alff_reho_surface(output_path, filename, bold_file):
+    """
+    Plot ReHo and ALFF for ciftis on surface.
+
+    Parameters
+    ----------
+    output_path :
+        path to save svg plot
+    filename :
+        surface file
+    bold_file :
+        original input bold file
+
+    Returns
+    ----------
+    output_path:
+        path to svg plot
+
+    """
+    import os
+
+    import matplotlib.pyplot as plt
+    import nibabel as nb
+    from bids.layout import parse_file_entities
+    from nilearn import plotting as plott
+    from templateflow.api import get as get_template
+
+    from xcp_d.utils.plot import surf_data_from_cifti
+
+    density = parse_file_entities(bold_file).get("den", "32k")
+    if density == "91k":
+        density = "32k"
+    rh = str(
+        get_template(template='fsLR', hemi="R", density="32k",
+                     suffix="midthickness", extension=".surf.gii")
+    )
+    lh = str(
+        get_template(template='fsLR', hemi="L", density="32k",
+                     suffix="midthickness", extension=".surf.gii")
+    )
+
+    cifti = nb.load(filename)
+    cifti_data = cifti.get_fdata()
+    cifti_axes = [cifti.header.get_axis(i) for i in range(cifti.ndim)]
+
+    fig, axes = plt.subplots(figsize=(16, 16), ncols=2, nrows=2, subplot_kw={'projection': '3d'})
+    output_path = os.path.abspath(output_path)
+    v_max = nb.load(filename).get_fdata().max()
+    v_min = nb.load(filename).get_fdata().max()
```

Comment:
```suggestion
    v_max = np.max([np.max(lh_surf_data), np.max(rh_surf_data)])
    v_min = np.min([np.min(lh_surf_data), np.min(rh_surf_data)])
```

Just so it only takes the surface data into account. The lines would need to go after the surface data are loaded though.